### PR TITLE
[api-extractor] allow packages to not have a version field

### PR DESCRIPTION
**Pull request by alan-agius4**
NPM does now allow that, this is actually a node implementation of modules. NPM will only validate the primary package.json and fail on publish if version is missing.

In the node documentation for modules it is clearly specified the the only requirement is the 'main' property.

Closes: #1044

### CODE REVIEW

**Comment by alan-agius4** [comment]
Not sure what’s the failure all out.

Also is there any contribution guidelines such as how to tess locally etc..?

File: libraries/node-core-library/src/test/test-data/example-package-no-version/src/ExampleFile.txt [hunk 1]
```diff
@@ -0,0 +1,29 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
```
Comment (iclanton) [inline libraries/node-core-library/src/test/test-data/example-package-no-version/src/ExampleFile.txt]: What is this file for?

**Comment by iclanton** [comment]
To test locally, make sure you have `@microsoft/rush` installed globally (`npm install -g @microsoft/rush`), install repo dependencies by running `rush update`, and build the repo by running `rush build`.

The tests are Jest tests, so you can debug them like any other Jest test.

**Comment by iclanton** [comment]
@octogonz - what are your thoughts on making the `version` field optional? I don't really have a strong opinion.

File: libraries/node-core-library/src/test/test-data/example-package-no-version/src/ExampleFile.txt (hunk 1 above)
Comment (alan-agius4) [reply libraries/node-core-library/src/test/test-data/example-package-no-version/src/ExampleFile.txt]: Will remove

**Comment by alan-agius4** [comment]
@iclanton you prefer a blank version example `0.0.0` instead of making it optional?

**Comment by alan-agius4** [comment]
@iclanton you prefer a blank version example `0.0.0` instead of making it optional?

**Comment by alan-agius4** [comment]
@pgonzal, I think there are some missing changelog entries.

**Comment by octogonz** [comment]
Yeah I wanted to talk to @iclanton about the API change first. It affects projects outside this repo. I'm thinking we might want to go the other way and rename the interface whose version is optional.

**Comment by alan-agius4** [comment]
Thanks for the quick update @octogonz.

**Comment by octogonz** [comment]
@alan-agius4 I updated the PR with the other approach we discussed.  I think it's much better.  This PR is pretty much good to go.  If someone could retest @shlomiassaf 's original repro from https://github.com/Microsoft/web-build-tools/issues/1044 , that would be great.

**Comment by iclanton** [comment]
Testing this real quick, but otherwise looks good.

**Pull request merged**